src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupService.scala

```diff
@@ -84,10 +84,45 @@ class ManagedGroupService(private val resourceService: ResourceService, private
       _ <- resourceService.deleteResource(Resource(managedGroupType.name, groupId))
     } yield ()
   }
+
+  def listAdminEmails(resourceId: ResourceId): Future[Set[WorkbenchEmail]] = {
+    listPolicyMemberEmails(resourceId, AccessPolicyName(ManagedGroupService.adminValue))
+  }
+
+  def listMemberEmails(resourceId: ResourceId): Future[Set[WorkbenchEmail]] = {
+    listPolicyMemberEmails(resourceId, AccessPolicyName(ManagedGroupService.memberValue))
+  }
+
+  private def listPolicyMemberEmails(resourceId: ResourceId, policyName: AccessPolicyName) = {
+    val resourceAndPolicyName = ResourceAndPolicyName(Resource(ManagedGroupService.managedGroupTypeName, resourceId), policyName)
+    accessPolicyDAO.loadPolicy(resourceAndPolicyName) flatMap {
+      case Some(policy) => directoryDAO.loadSubjectEmails(policy.members)
+      case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"Group or policy could not be found: $resourceAndPolicyName"))
+    }
+  }
+
+  def overwriteAdminEmails(resourceId: ResourceId, emails: Set[WorkbenchEmail]): Future[AccessPolicy] = {
+    overwritePolicyMemberEmails(resourceId, AccessPolicyName(ManagedGroupService.adminValue), emails)
+  }
+
+  def overwriteMemberEmails(resourceId: ResourceId, emails: Set[WorkbenchEmail]): Future[AccessPolicy] = {
+    overwritePolicyMemberEmails(resourceId, AccessPolicyName(ManagedGroupService.memberValue), emails)
+  }
+
+  private def overwritePolicyMemberEmails(resourceId: ResourceId, policyName: AccessPolicyName, emails: Set[WorkbenchEmail]): Future[AccessPolicy] = {
+    val resourceAndPolicyName = ResourceAndPolicyName(Resource(ManagedGroupService.managedGroupTypeName, resourceId), policyName)
+    accessPolicyDAO.loadPolicy(resourceAndPolicyName).flatMap {
+      case Some(policy) => val updatedPolicy = AccessPolicyMembership(emails, policy.actions, policy.roles)
+        resourceService.overwritePolicy(managedGroupType, policyName, resourceAndPolicyName.resource, updatedPolicy)
+      case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"Group or policy could not be found: $resourceAndPolicyName"))
+    }
+  }
 }
 
 object ManagedGroupService {
-  val memberRoleName = ResourceRoleName("member")
+  val memberValue = "member"
+  val adminValue = "admin"
```

Comment:
I don't really like have these Strings floating around publicly. I would have a ResourceRoleName and AccessPolicyName val for admin and member and use those throughout. If you want them to use the same private string val, great.

Reply:
fixed